core/src/components/datetime/datetime.tsx

```diff
@@ -1196,14 +1178,22 @@ export class Datetime implements ComponentInterface {
     });
   }
 
-  private processValue = (value?: string | string[] | null) => {
-    const hasValue = value !== null && value !== undefined;
+  private processValue = async (value?: string | string[] | null, animate = false) => {
+    const hasValue = value !== null && value !== undefined && (!Array.isArray(value) || value.length > 0);
```

Comment:
This change avoids a crash when manually setting a multiple selection datetime's value to `[]`, by treating an empty array the same as setting the value to `undefined` (i.e. clearing the selected dates). This is done by the `multiple` test if the fixture's `goto` is called without passing in a default value.

I believe we weren't seeing the test crash before because `processValue` wasn't used when updating the value async, although I believe there was still a bug present, just with stricter circumstances.